### PR TITLE
Resolve #7

### LINKED ISSUE #7: As a developer, I want to have java template repo
## 💪 Motivation
...so that I can kick-off a project with all the things the PDS expects as part of the repo. (and it builds with CI/CD too!)

## 📖 Additional Details

See https://github.com/NASA-PDS/pds-template-repo-python/issues/13 for some additional things we want to add as well. Basically, we want to have a stable template java repo.

- [x] Add a template "Hello World"-like doc source page so the mvn site can be built via CI/CD
- [x] Create a Github Pages branch to push the docs to as part of CI/CD
- [x] Add 1 template java test so mvn test can successfully run via CI/CD (e.g. `sys.exit(0)`)
- [x] Update default docs to look similar to NASA-PDS/pds-template-repo-python
- [x] Update stable_cicd.yaml to use latest roundup (e.g. triggered by `release/*` branch)
- [x] Add CONTRIBUTOR guide - [github says some things here](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file) but open to something apache-like
- [x] Add CODE_OF_CONDUCT - [example from some SMD folks](https://github.com/spacetelescope/style-guides/blob/master/CODE_OF_CONDUCT.md), but open to something apache-like

### PR DESCRIPTION
**Pull request by nutjob4life**
## 📜 Summary

Merge this if you dare and resolve #7, which includes:

- Template docs that can be expanded upon so you get a default site
- Template "Hello World" app that uses resources in a jar so you get to exercise jar construction with non-compiled files
- Javadocs to populate the site reports including the package-level docs that everyone forgets
- Template unit test that can be modified and expanded upon
- (The gh-pages branch was already there; see #7)
- Site look-and-feel like other PDS repositories
- Stabel CI/CD workflow
- Updated documentation throughout
- A contributor's guide
- A code of conduct

##  🩺 Test Data and/or Report

```console
$ mvn test
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
…
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
…
```

## 🧩 Related Issues

#7

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        </resources>
+
+        <!--
+        Plugins tell Maven how to operate the various lifecycle phases; include extra as needed, such
+        as for CORBA stub generation from `.idl` files, or even more modern things.
+        -->
+        <plugins>
+            <!-- This plugin actually compiles the `.java` source code, so you probably need it! -->
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (jordanpadams) [inline pom.xml:137]: @nutjob4life if we use pds parent, then i don't think we need this?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                <version>3.8.1</version>
+                <configuration>
+                    <useIncrementalCompilation>false</useIncrementalCompilation>
+                </configuration>
+            </plugin>
+            <!-- This plugin runs the testing framework, so you also probably need it -->
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (jordanpadams) [inline pom.xml:145]: @nutjob4life if we use pds parent, then i don't think we need this?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    <developers/>
+    <profiles/>
+    <distributionManagement/>
+    <reporting/>
+    <licenses/>
+    <!--
+    The <organization> should be inherited from the parent but isn't there yet, so we include it explicitly
+    here in this template `pom.xml`. Of course, you'll still be able to override it.
+    -->
+    <organization>
```
Comment (jordanpadams) [inline pom.xml:181]: @nutjob4life if we use pds parent, then i don't think we need this?

**Review by jordanpadams — CHANGES_REQUESTED** [review]
@nutjob4life can we use the maven parent here?

File: pom.xml (hunk 1 above)
Comment (nutjob4life) [reply pom.xml]: Included in 52b4149f5be360ad91b83281190013aa681a9e55

File: pom.xml (hunk 2 above)
Comment (nutjob4life) [reply pom.xml]: Included in 52b4149f5be360ad91b83281190013aa681a9e55

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        <project.name>myproject</project.name>
+
+        <!-- You'll want a Unicode-type encoding here in almost every case -->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <!-- Adjust as needed -->
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>${maven.compiler.source}</maven.compiler.target>
+    </properties>
+
```
[the remark this answers is not in the document]
Comment (nutjob4life) [reply pom.xml:51]: Included in 52b4149f5be360ad91b83281190013aa681a9e55

File: pom.xml (hunk 3 above)
Comment (nutjob4life) [reply pom.xml]: Just don't forget to add `<organization>` to `gov:nasa:pds:1.9.0:pom` 👍 

Removed `<organization>` from here in 52b4149f5be360ad91b83281190013aa681a9e55

**Review by jordanpadams — APPROVED** [review]

**Pull request merged**